src/GafferArnold/IECoreArnoldPreview/Renderer.cpp

```diff
@@ -1241,6 +1199,90 @@ class ArnoldAttributes : public IECoreScenePreview::Renderer::AttributesInterfac
 
 		};
 
+	struct Volume
+	{
+		Volume( const IECore::CompoundObject *attributes )
+		{
+			volumeGrids = attribute<IECore::StringVectorData>( g_volumeGridsAttributeName, attributes );
+			velocityGrids = attribute<IECore::StringVectorData>( g_velocityGridsAttributeName, attributes );
+			velocityScale = attributeValue<float>( g_velocityScaleAttributeName, attributes, 1.0f );
+			velocityFPS = attributeValue<float>( g_velocityFPSAttributeName, attributes, 24.0f );
+			velocityOutlierThreshold = attributeValue<float>( g_velocityOutlierThresholdAttributeName, attributes, 0.001f );
+		}
+
+		IECore::ConstStringVectorDataPtr volumeGrids;
+		IECore::ConstStringVectorDataPtr velocityGrids;
+		float velocityScale;
+		float velocityFPS;
+		float velocityOutlierThreshold;
+
+		void hash( IECore::MurmurHash &h ) const
+		{
+			if( volumeGrids )
+			{
+				volumeGrids->hash( h );
+			}
+
+			if( velocityGrids )
+			{
+				velocityGrids->hash( h );
+			}
+
+			h.append( velocityScale );
+			h.append( velocityFPS );
+			h.append( velocityOutlierThreshold );
+		}
+
+		void apply( AtNode *node ) const
+		{
+			if( volumeGrids && volumeGrids->readable().size() )
+			{
+				AtArray *array = ParameterAlgo::dataToArray( volumeGrids.get(), AI_TYPE_STRING );
+				AiNodeSetArray( node, g_volumeGridsArnoldString, array );
+			}
+
+			if( velocityGrids && velocityGrids->readable().size() )
+			{
+				AtArray *array = ParameterAlgo::dataToArray( velocityGrids.get(), AI_TYPE_STRING );
+				AiNodeSetArray( node, g_velocityGridsArnoldString, array );
+			}
+
+			if( velocityScale != 1.0f && AiNodeEntryLookUpParameter( AiNodeGetNodeEntry( node ), g_velocityScaleArnoldString ) )
```

Comment:
I think this `AiNodeLookUpParameter()` call is redundant now we know we're only operating on volume nodes, isn't it?